### PR TITLE
Psi

### PR DESCRIPTION
**Pull request by mike-coolfront**
psi branch augments node-quickbooks to allow us the ability to debug in projection mode as well
as supply custom logging functions that we can funnel log data to our own mechanisms

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -47,6 +49,8 @@ function QuickBooks(consumerKey, consumerSecret, token, tokenSecret, realmId, us
   this.realmId         = eval(prefix + 'realmId')
   this.useSandbox      = eval(prefix + 'useSandbox')
   this.debug           = eval(prefix + 'debug')
+  this.logFunc         = logFunc || console.log
```
Comment (aaron-coolfront) [inline index.js]: Should we do anything there to ensure that 'logFunc' is an actual function or is this being too paranoid?

File: index.js [hunk 2]
```diff
@@ -47,6 +49,8 @@ function QuickBooks(consumerKey, consumerSecret, token, tokenSecret, realmId, us
...
+  this.requestLogFunc  = requestLogFunc
```
Comment (aaron-coolfront) [inline index.js]: Should we do anything there to ensure that 'requestLogFunc' is an actual function or is this being too paranoid?

**Review by aaron-coolfront — COMMENTED** [review]
Just a couple of questions.

File: index.js (hunk 2 above)
Comment (mike-coolfront) [reply index.js]: I think its a good idea.

File: index.js (hunk 1 above)
Comment (mike-coolfront) [reply index.js]: good idea

**Review by aaron-coolfront — COMMENTED** [review]
Looks good

**Pull request merged**